src/tokens/ERC1155/ERC1155Sale.sol

```diff
@@ -0,0 +1,296 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.17;
+
+import {IERC1155Sale} from "./IERC1155Sale.sol";
+import {ERC1155, ERC1155Meta} from "@0xsequence/erc-1155/contracts/tokens/ERC1155/ERC1155Meta.sol";
+import {ERC1155Metadata} from "@0xsequence/erc-1155/contracts/tokens/ERC1155/ERC1155Metadata.sol";
+import {ERC1155Supply} from "./ERC1155Supply.sol";
+import {ERC1155SaleErrors} from "./ERC1155SaleErrors.sol";
+import {ERC2981} from "@openzeppelin/contracts/token/common/ERC2981.sol";
+import {AccessControl} from "@openzeppelin/contracts/access/AccessControl.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+contract ERC1155Sale is
+    IERC1155Sale,
+    ERC1155SaleErrors,
+    ERC1155Supply,
+    ERC1155Meta,
+    ERC1155Metadata,
+    ERC2981,
+    AccessControl
+{
+    bytes32 public constant MINT_ADMIN_ROLE = keccak256("MINT_ADMIN_ROLE");
+    bytes32 public constant ROYALTY_ADMIN_ROLE = keccak256("ROYALTY_ADMIN_ROLE");
+
+    bytes4 private constant ERC20_TRANSFERFROM_SELECTOR =
+        bytes4(keccak256(bytes("transferFrom(address,address,uint256)")));
+
+    bool private _initialized;
+
+    // ERC20 token address for payment. address(0) indicated payment in ETH.
+    address private _paymentToken;
+
+    SaleDetails private _globalSaleDetails;
```

Comment:
IMHO it's better to only prefix with `_` if we are talking about a function's argument